### ISSUE TITLE
search_author didn't work

### ISSUE THREAD

**Issue by trioputrap** [OrganicIrradiation/scholarly#27]
Traceback (most recent call last):
  File "~project/test.py", line 4, in <module>
    print(next(search_query))
  File "~project\venv\lib\site-packages\scholarly.py", line 99, in _search_citation_soup
    yield Author(tablerow)
  File "~project\venv\lib\site-packages\scholarly.py", line 209, in __init__
    self.name = __data.find('h3', class_='gsc_1usr_name').text
AttributeError: 'NoneType' object has no attribute 'text'

Got this error when I try to search author, but search publication still work. Please fix this issue, thankyou

**Comment by zdw890828** [comment]
Find the same problems when running my code:
  File "~Program\venv\lib\site-packages\scholarly.py", line 112, in _search_citation_soup
    yield Author(row)
  File "~Program\venv\lib\site-packages\scholarly.py", line 228, in __init__
    self.name = __data.find('h3', class_='gsc_oai_name').text
AttributeError: 'NoneType' object has no attribute 'text'

**Comment by bondankit07** [comment]
`search_author` is not working.

```
Traceback (most recent call last):
  File "citation_report_google.py", line 7, in <module>
    author = next(search_query).fill()
  File "/home/lab1/.local/lib/python2.7/site-packages/scholarly/scholarly.py", line 110, in _search_citation_soup
    yield Author(row)
  File "/home/lab1/.local/lib/python2.7/site-packages/scholarly/scholarly.py", line 226, in __init__
    self.name = __data.find('h3', class_='gsc_oai_name').text
AttributeError: 'NoneType' object has no attribute 'text'
```

**Comment by stevenwaslander** [comment]
Confirmed, same issue.

**Comment by jweob** [comment]
Created a pull request to fix this here https://github.com/OrganicIrradiation/scholarly/pull/28

My first ever open source pull request, be kind!

**Comment by trioputrap** [comment]
> 
> 
> Created a pull request to fix this here #28
> 
> My first ever open source pull request, be kind!

thankyou, it works on me

**Comment by arranger1044** [comment]
@jweob I tried your branch but it is not working for me, am I missing something?

**Comment by adinriv** [comment]
Similar to the PR #28, it seems that the problem lies with the changes on the IDs of the main page.  The `self.name` and the `affiliation` on the constructor of `Author` class changed.  If you update to:

```python
self.name = __data.find('h3', class_='gs_ai_name').find('a').text
affiliation = __data.find('div', class_='gs_ai_aff')
```
plus the changes on #28 (which seem to be already incorporated) seem to be working now.

**Comment by milanjanosov** [comment]
Does anyone currently has a working version? Thanks!

**Comment by RogerCYoung176** [comment]
> Does anyone currently have a working version? Thanks!

It's easy to resolve as per adinriv comment on 3 Apr. If you click the line reference it will take you to where you need to amend the code.

I'm fairly new to this, but it was pretty straight forward, the error message read:
> File "Your Dir", line 226, in \__init__\
> self.name = __data.find('h3', class_='gsc_oai_name').text
> AttributeError: 'NoneType' object has no attribute 'text'

Just follow the link by clicking line 226 in the console and it should take you toscholarly.py file and replace the code as per below.

Original code:
> self.name = __data.find('h3', class_='gsc_oai_name').text 
> affiliation = __data.find('div', class_='gs_ai_aff') 

Replacement code (as per adinriv comment on 3 Apr):
> self.name = __data.find('h3', class_='gs_ai_name').find('a').text
> affiliation = __data.find('div', class_='gs_ai_aff')

Works fine after this with no errors and takes 2 mins.

**Comment by OrganicIrradiation** [comment]
Thanks for the PR @jweob! I ended up merging in PR #40, which fixes this issue in a more dynamic way. Could you please confirm that the current codebase works for you? If so, I'll need to push the updated version to PyPI.

**Comment by bielsnohr** [comment]
Assumed fixed by #40
